### ISSUE TITLE
Not compiling on Windows, 1.24 nightly

### ISSUE THREAD

**Issue by regresscheck** [tomusdrw/rust-web3#81]
Hi.
I'm getting this when trying to build example:
```
error[E0432]: unresolved import `self::tokio_uds::UnixStream`
  --> C:\Users\Andrey\.cargo\git\checkouts\rust-web3-ebc1a09fd5a9a2a2\c6bd1ab\src\transports\ipc.rs:10:5
   |
10 | use self::tokio_uds::UnixStream;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `UnixStream` in the root

error: aborting due to previous error

error: Could not compile `web3`.

```

Tried both gnu and msvc toolchains, cargo and github versions. Am I missing something?

**Comment by tomusdrw** [comment]
IPC feature is enabled by default, but is not available on windows. Could you try reference the library like this:
```
web3 = { git = "https://github.com/tomusdrw/rust-web3.git", default-features = false, features = ["http"] }
```

**Comment by regresscheck** [comment]
Thanks! Everything is fine now. Can you please mention it somewhere, so I would be the only one creating such issue? :)

**Comment by tomusdrw** [comment]
Answers are now in this thread:) I would really appreciate a PR to the readme file mentioning windows build though.

**Comment by afck** [comment]
This way, every dependent crate has to do a lot of conditional compilation and have a complicated `Cargo.toml` if it wants to offer IPC support on Unix but also run on Windows. Wouldn't it make more sense if `web3` itself compiled on Windows, but `Ipc::new` just always failed with an error message explaining that this is Unix-only? That would just involve a little bit more conditional compilation in `transports/mod.rs` and an `ipc_windows` dummy module that would do nothing expect returning that error.

If you are okay with this, please reopen this Issue, then I'll try and write the code.

Alternatively, the `ipc` module could just be omitted on Windows (but without failing to compile). That would still be less complicated for the user than it is now, because at least the `Cargo.toml` could just keep the `ipc` feature unconditionally.

**Comment by tomusdrw** [comment]
@afck both solutions make sense to me, feel free to submit a PR

**Comment by tomusdrw** [comment]
Fixed in #124
